sys/riscv/riscv/freebsd64_machdep.c

```diff
@@ -0,0 +1,190 @@
+ * All rights reserved.
+ * Copyright (c) 2016 Yukishige Shibata <y-shibat@mtd.biglobe.ne.jp>
+ * All rights reserved.
+ * Copyright (c) 2020 John Baldwin
+ *
+ * Portions of this software were developed by SRI International and the
+ * University of Cambridge Computer Laboratory under DARPA/AFRL contract
+ * FA8750-10-C-0237 ("CTSRD"), as part of the DARPA CRASH research programme.
+ *
+ * Portions of this software were developed by the University of Cambridge
+ * Computer Laboratory as part of the CTSRD Project, with support from the
+ * UK Higher Education Innovation Fund (HEIF).
+ *
+ * Portions of this software were developed by SRI International and the
+ * University of Cambridge Computer Laboratory (Department of Computer
+ * Science and Technology) under DARPA contract HR0011-18-C-0016 ("ECATS"),
+ * as part of the DARPA SSITH research programme.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#define __ELF_WORD_SIZE 64
+#define	EXPLICIT_USER_ACCESS
+
+#include <sys/cdefs.h>
+__FBSDID("$FreeBSD$");
+
+#include <sys/param.h>
+#include <sys/systm.h>
+#include <sys/imgact.h>
+#include <sys/imgact_elf.h>
+#include <sys/kernel.h>
+#include <sys/proc.h>
+#include <sys/syscallsubr.h>
+#include <sys/sysent.h>
+#include <sys/ucontext.h>
+
+#include <machine/md_var.h>
+
+#include <compat/freebsd64/freebsd64_proto.h>
+#include <compat/freebsd64/freebsd64_syscall.h>
+#include <compat/freebsd64/freebsd64_util.h>
+
+static void	freebsd64_sendsig(sig_t, ksiginfo_t *, sigset_t *);
+
+extern const char *freebsd64_syscallnames[];
+
+struct sysentvec elf_freebsd_freebsd64_sysvec = {
+	.sv_size	= FREEBSD64_SYS_MAXSYSCALL,
+	.sv_table	= freebsd64_sysent,
+	.sv_errsize	= 0,
```

Comment:
I'm not entirely sure it makes sense to include the explicit 0/NULL initializers.

Reply:
I don't either, I was just copying what was in sys/riscv/riscv/elf_machdep.c and stayed in sync with that.  We could fix elf_machdep.c upstream perhaps.